bin/strata-sequencer-client/src/main.rs

```diff
@@ -70,13 +72,21 @@ fn main_inner(args: Args) -> Result<()> {
     );
     executor.spawn_critical_async(
         "duty-runner",
-        duty_executor_worker(rpc, duty_rx, handle.clone(), idata),
+        duty_executor_worker(rpc, duty_rx, handle.clone(), idata_clone),
     );
 
     task_manager.start_signal_listeners();
-    task_manager.monitor(Some(Duration::from_millis(SHUTDOWN_TIMEOUT_MS)))?;
+    let result = task_manager.monitor(Some(Duration::from_millis(SHUTDOWN_TIMEOUT_MS)));
 
-    Ok(())
+    // Zeroize idata
+    if let Ok(mut idata) = Arc::try_unwrap(idata) {
```

Comment:
Maybe we won't be doing `Arc::try_unwrap` for `idata` in other places but as per the docs, it is strongly recommended to use `Arc::into_inner` if the `Err` variant is not used(like in this case), to prevent potential race conditions and having to lose the value.

https://doc.rust-lang.org/std/sync/struct.Arc.html#method.try_unwrap

Reply:
TIL!